### PR TITLE
Pass date and permit type to spider

### LINKED ISSUE #2: Add a new callback for Commercial Wrecking requests
Extend the Scrapy chain to make a request for Commercial Wrecking Permit requests and parse the results. Issue #1 is to write the method to parse the table, this issue is to make the second request.

Ideally, the commercial wrecking permit results should be stored in the same python dict as the residential requests.

### PR DESCRIPTION
**Pull request by thashepherd**
Resolves https://github.com/codefordayton/demolition_checker/issues/2.

This PR extracts the spider with @aldrio's record extraction code to its own file, and sets it up to receive arguments from `main.py`.

Within the `crawl` method in `main.py`, we can write:
```
    yield runner.crawl(
        DemolitionSpider,
        permit_type=PermitType.commercial_wrecking_permit,
        start_date="08/01/2023",
    )
```
...which will run the spider with that permit type and start date. Note that `start_date` is a str and that I haven't included any sort of date parsing or validation. Also note the boilerplate code in `main.py` which is necessary to run multiple spiders and wait for them all to finish.

When running this script, you can now use `python main.py`. You can still run `scrapy runspider src/demolition_spider.py` as well.

### CODE REVIEW

File: src/demolition_spider.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+            # select the data we want from the table row
+            record_number = row.xpath("string(.//td[2])").get().strip()
+            record_link = row.xpath(".//td[2]//a/@href").get()
+            record_type = row.xpath("string(.//td[3])").get().strip()
+            project_name = row.xpath("string(.//td[4])").get().strip()
+            address = row.xpath("string(.//td[5])").get().strip()
+
+            # make sure that the data is as expected
+            assert record_number, "Record number not found"
```
Comment (DavidEBest) [inline src/demolition_spider.py]: Could you update this logic to the approach @aldrio took - replacing asserts with throws?

Comment (thashepherd) [reply src/demolition_spider.py]: Yeah, just took care of that. Was based on an older version of his PR.

Some of this logic will be impacted by https://github.com/thashepherd/demolition_checker/pull/1 anyway :)

**Comment by thashepherd** [comment]
Updated - should merge cleanly now.

File: src/demolition_spider.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                'ctl00$PlaceHolderMain$generalSearchForm$txtGSStartDate': self.start_date,
+        }
+        print(f"Setting the following formdata:\n{form_data}")
+        form_data['__VIEWSTATE'] = response.css('input#__VIEWSTATE::attr(value)').extract_first()
+
+        yield FormRequest.from_response(
+            response,
+            formdata = form_data,
+            clickdata={'id': 'ctl00_PlaceHolderMain_btnNewSearch'},
+            callback=self.parseResults)
```
Comment (thashepherd) [inline src/demolition_spider.py:39]: The changes I made to the spider are contained within these lines:

- PermitType enum
- Additional class properties & init method for DemolitionSpider
- Mild refactoring of form_data creation

**Pull request merged**